### PR TITLE
Add README for `packages`

### LINKED ISSUE #3383: Explain how the packages are associated with each other in the Meta Package
**Describe the feature request**
- Explain how the packages are associated with each other in the Meta Package for the packages under `./packages` directory. 

**Describe alternative(s) you've considered**

**Affected product area (please put an X in all that apply)**

- ( ) APIs
- ( ) Addons
- ( ) CLI
- (x) Docs
- ( ) IAM
- ( ) Installation
- ( ) Plugin
- ( ) Security
- ( ) Test and Release
- ( ) User Experience
- ( ) Developer Experience

**Additional context**

### PR DESCRIPTION
**Pull request by anujc25**
### What this PR does / why we need it
- Add README for `packages` to explain how the packages are associated with each other in the Meta Package

### Which issue(s) this PR fixes

Fixes #3383

### Describe testing done for PR
None

### Release note

```release-note
None
```

### CODE REVIEW

File: packages/README.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+- cluster-api-bootstrap-kubeadm
+- cluster-api-control-plane-kubeadm
+- cluster-api-provider-aws
+- cluster-api-provider-azure
+- cluster-api-provider-docker
+- cluster-api-provider-vsphere
+
+## Standalone Packages
+
+This type of carvel packages are individual packages that doesn't depend on other packages. Below are the list of standalone packages:
```
Comment (marckhouzam) [inline packages/README.md:42]: "that doesn't" -> "that don't"
"Below are the list" -> "Below is the list"

File: packages/README.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+- standalone-plugins
+- capabilities
+- tkg-autoscaler
```
Comment (marckhouzam) [inline packages/README.md:46]: Should this be "tkg-storageclass" instead?

File: packages/README.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
+# Packages
+
+This directory contains two types of packages. Management Packages and Standalone Packages
+
+## Management Packages
+
+This directory contains some carvel packages for management components. Any package that performs the role of managing and operating a workload cluster is a Management package. This management packages are purpose-built to be installed on the management cluster. More details about packages can be found [here](../docs/packages/definitions.md)
```
Comment (marckhouzam) [inline packages/README.md:7]: "This management packages" -> "These management packages"

File: packages/README.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+This management packages are associated with each in terms of [Meta Package](../docs/packages/definitions.md#meta-package). A Meta Package contains PackageInstall CRs, Secrets and ServiceAccounts, RBAC rules that are needed to install other packages and when the meta package is installed, it installs the other packages.
```
Comment (marckhouzam) [inline packages/README.md:9]: "This management packages" -> "These management packages"

There seems to be a word missing in "This management packages are associated with each in terms".  With "each" what?

File: packages/README.md [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    - cliplugins
+  - tkg-clusterclass (TKG ClusterClass Meta Package)
+    - tkg-clusterclass-docker
+    - tkg-clusterclass-aws
+    - tkg-clusterclass-azure
+    - tkg-clusterclass-vsphere
+  - core-management-plugins
+  - tkr-source-controller
+
+As part of the Bootstrap and Management Cluster creation, `management-cluster` plugin installs `tkg` meta package which installs all child packages. As part of this package installation `management-cluster` plugin creates [values.yaml](./tkg/bundle/config/values.yaml) file based on the user input and the configuration is propogated to child packages based on overlays written within the package.
```
Comment (marckhouzam) [inline packages/README.md:28]: - "`management-cluster` plugin installs `tkg`" -> "the `management-cluster` plugin installs the `tkg`" (two extra "the")
- "package installation `management-cluster` plugin" -> "package installation the `management-cluster` plugin"
- "`management-cluster` plugin create values.yaml" -> "`management-cluster` plugin create a values.yaml"
- "based on the user input and the configuration is propagated" -> "based on the user input.  The configuration is then propagated"

File: packages/README.md [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+There are few management packages created under this directory but not getting used at the moment and not intgrated with `tkg` meta package. Below is the list of these packages:
```
Comment (marckhouzam) [inline packages/README.md:30]: "but not getting used at the moment" -> "which are not used at the moment"
"intgrated with `tkg`" -> "integrated with the `tkg`" (typo and extra "the")

**Review by marckhouzam — COMMENTED** [review]
Some suggestions

File: packages/README.md (hunk 2 above)
Comment (anujc25) [reply packages/README.md:46]: Added `tkg-storageclass` as well along with `tkg-autoscaler` as both looks like standalone packages at the moment.

File: packages/README.md [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+    - cliplugins
+  - tkg-clusterclass (TKG ClusterClass Meta Package)
+    - tkg-clusterclass-docker
+    - tkg-clusterclass-aws
+    - tkg-clusterclass-azure
+    - tkg-clusterclass-vsphere
+  - core-management-plugins
+  - tkr-source-controller
+
+As part of the Bootstrap and Management Cluster creation, the `management-cluster` plugin installs the `tkg` meta package which installs all child packages. As part of this package installation the `management-cluster` plugin create [values.yaml](./tkg/bundle/config/values.yaml) file based on the user input. The configuration is then propogated to child packages based on overlays written within the package.
```
Comment (marckhouzam) [inline packages/README.md:28]: I missed a couple before:
- "create" -> "creates a"
- "propogated" -> "propagated"

**Review by marckhouzam — APPROVED** [review]
I missed a couple but after those two. it LGTM

**Review by vuil — APPROVED** [review]
lgtm, thanks!

**Pull request merged**